libtransact/protos/transaction_receipt.proto

```diff
@@ -18,15 +18,34 @@ syntax = "proto3";
 import "events.proto";
 
 message TransactionReceipt {
+  enum Result {
+    VALID = 0;
```

Comment:
Do we want the default to be VALID?  We usally have an enum value like "UNSET_RESULT = 0" for protobuf enums

Reply:
Updated

Reply:
Actually, I'm not sure how to handle this, since it's breaks compatibility with Sawtooth 1.0, and therefore causes the `protocol::receipt::tests::transaction_receipt_sawtooth10_compatibility` test to fail.

Reply:
Updated to interpret an unset result as VALID when parsing a protobuf as a native Rust struct; this maintains compatibility with Sawtooth 1.x